### PR TITLE
[SPG]:216 terraform support for environment group

### PR DESCRIPTION
**Pull request by RajBaviskar**
* Adding Terraform support for EnvironmenGroup API

### CODE REVIEW

File: internal/service/platform/environment_group/data_source_environment_group.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+			},
+			"project_id": {
+				Description: "project_id of the environment group.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"color": {
+				Description: "Color of the environment group.",
+				Type:        schema.TypeString,
+				Optional:    true,
```
Comment (rathodmeetsatish) [inline internal/service/platform/environment_group/data_source_environment_group.go:40]: computed  =true

File: internal/service/platform/environment_group/data_source_environment_group.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+			},
+			"yaml": {
+				Description: "Input Set YAML",
+				Type:        schema.TypeString,
+				Required:    true,
```
Comment (rathodmeetsatish) [inline internal/service/platform/environment_group/data_source_environment_group.go:45]: computed = true

File: internal/service/platform/environment_group/data_source_environment_group_test.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+			                 projectIdentifier: ${harness_platform_project.test.id}
+			                 envIdentifiers: []
+		  EOT
+		}
+
+		data "harness_platform_environment_group" "test" {
+			identifier = "%[1]s"
+			org_id = harness_platform_environment_group.test.org_id
+			project_id = harness_platform_environment_group.test.project_id
+			yaml = <<-EOT
```
Comment (rathodmeetsatish) [inline internal/service/platform/environment_group/data_source_environment_group_test.go:69]: we dont need to add yaml just identifier,project_id,org_id is enough

File: internal/service/platform/environment_group/resource_environment_group_test.go [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+					resource.TestCheckResourceAttr(resourceName, "org_id", id),
+					resource.TestCheckResourceAttr(resourceName, "project_id", id),
+					resource.TestCheckResourceAttr(resourceName, "color", updatedColor),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{"yaml"},
```
Comment (rathodmeetsatish) [inline internal/service/platform/environment_group/resource_environment_group_test.go:51]: why do we need to do this?

File: internal/service/platform/environment_group/resource_environment_group_test.go [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+					ProjectIdentifier := id
+
+					resp, _, err := c.EnvironmentGroupApi.DeleteEnvironmentGroup(ctx, id, c.AccountId, OrgIdentifier, ProjectIdentifier, &nextgen.EnvironmentGroupApiDeleteEnvironmentGroupOpts{
+						Branch:         optional.NewString(id),
+						RepoIdentifier: optional.NewString(id),
+					})
+					require.NoError(t, err)
+					require.True(t, resp.Data.Deleted)
+				},
+				Config:             testAccResourceEnvironmentGroup(id, name, color),
```
Comment (rathodmeetsatish) [inline internal/service/platform/environment_group/resource_environment_group_test.go:91]: import should we similar to above method

File: internal/service/platform/environment_group/data_source_environment_group.go (hunk 1 above)
Comment (RajBaviskar) [reply internal/service/platform/environment_group/data_source_environment_group.go]: changed

File: internal/service/platform/environment_group/data_source_environment_group.go (hunk 2 above)
Comment (RajBaviskar) [reply internal/service/platform/environment_group/data_source_environment_group.go]: changed

File: internal/service/platform/environment_group/data_source_environment_group_test.go (hunk 3 above)
Comment (RajBaviskar) [reply internal/service/platform/environment_group/data_source_environment_group_test.go]: removed

File: examples/data-sources/harness_platform_environment_group/data-source.tf [hunk 6]
```diff
@@ -0,0 +1,4 @@
+data "harness_platform_environment_group" "test" {
+  org_id     = "org_id"
```
Comment (rathodmeetsatish) [inline examples/data-sources/harness_platform_environment_group/data-source.tf:2]: identifier should also be present

File: examples/resources/harness_environment_group/import.sh [hunk 7]
```diff
@@ -0,0 +1,2 @@
+# Import using the Harness application id and environment group id.
```
Comment (rathodmeetsatish) [inline examples/resources/harness_environment_group/import.sh:1]: i guess there is no need for harness apllication id

File: examples/resources/harness_environment_group/resource.tf [hunk 8]
```diff
@@ -0,0 +1,16 @@
+resource "platform_environment_group" "test" {
+  identifier = "identifier"
+  org_id     = "orgIdentifer"
+  project_id = "projectIdentifier"
+  color = "#0063F7"
+  yaml       = <<-EOT
+   ---
+  environmentGroup:
+    name: "%[1]s"
```
Comment (rathodmeetsatish) [inline examples/resources/harness_environment_group/resource.tf:9]: name: "name" similar change other fields

File: examples/data-sources/harness_platform_environment_group/data-source.tf (hunk 6 above)
Comment (RajBaviskar) [reply examples/data-sources/harness_platform_environment_group/data-source.tf:3]: added

File: examples/resources/harness_environment_group/import.sh (hunk 7 above)
Comment (RajBaviskar) [reply examples/resources/harness_environment_group/import.sh:1]: removed

File: examples/resources/harness_environment_group/resource.tf (hunk 8 above)
Comment (RajBaviskar) [reply examples/resources/harness_environment_group/resource.tf]: changed

File: examples/data-sources/harness_platform_environment_group/data-source.tf (hunk 6 above)
Comment (rathodmeetsatish) [reply examples/data-sources/harness_platform_environment_group/data-source.tf:3]: please remove from above comment as well

File: examples/resources/harness_environment_group/import.sh (hunk 7 above)
Comment (rathodmeetsatish) [reply examples/resources/harness_environment_group/import.sh:1]: # Import using the Harness application id and environment group id
please review from above comment as well

Comment (RajBaviskar) [reply examples/resources/harness_environment_group/import.sh]: changed.

File: docs/data-sources/platform_environment_group.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "harness_platform_environment_group Data Source - terraform-provider-harness"
+subcategory: ""
```
Comment (rathodmeetsatish) [inline docs/data-sources/platform_environment_group.md:4]: Next Gen

Comment (RajBaviskar) [reply docs/data-sources/platform_environment_group.md]: added

**Review by rathodmeetsatish — APPROVED** [review]

**Review by rijajoo — APPROVED** [review]

**Pull request merged**